packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,39 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target, relatedTarget }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const oldContains = relatedTarget !== this && this.contains(relatedTarget as DDSCard);
+
+    const currentChildIndex = Array.from(this.children).indexOf(target as HTMLElement);
+    const oldChildIndex = Array.from(this.children).indexOf(relatedTarget as HTMLElement);
+
+    const targetPageIndex = currentChildIndex % this.pageSize;
+
+    // reset to first page if tabbing back into the carousel after previously moving pages
+    if (currentContains && currentChildIndex === 0) {
+      this.start = 0;
+    }
+
+    if (currentContains && oldContains) {
+      // going forwards, change page if focused card is the first on the next page
+      if (targetPageIndex === 0) {
```

Comment:
This seems still "on the edge" detection, given it seems to detect if "focus is on first card on the page", right? Can we detect if the index of the focused card of out of the page range (larger than equal to `start` and less than `start + pageSize`) instead?